CactbotOverlay/CactbotEventSource.cs

```diff
@@ -370,145 +370,97 @@ protected override void Update()
-          var job = ffxiv_.GetWhiteMage();
-          if (job != null) {
-            if (send || !job.Equals(notify_state_.whm)) {
-              notify_state_.whm = job;
-              var e = new JSEvents.PlayerChangedEvent(player);
-              e.jobDetail = new JSEvents.PlayerChangedEvent.WhiteMageDetail(job);
-              OnPlayerChanged(e);
-            }
-          }
-        } else if (player.job == FFXIVProcess.EntityJob.SMN || player.job == FFXIVProcess.EntityJob.SCH || player.job == FFXIVProcess.EntityJob.ACN) {
-          var job = ffxiv_.GetSummonerAndScholar();
-          if (job != null) {
-            if (send || !job.Equals(notify_state_.smn_sch)) {
-              notify_state_.smn_sch = job;
-              var e = new JSEvents.PlayerChangedEvent(player);
-              e.jobDetail = new JSEvents.PlayerChangedEvent.SummonerAndScholarDetail(job);
-              OnPlayerChanged(e);
-            }
-          }
-        } else if (player.job == FFXIVProcess.EntityJob.MNK || player.job == FFXIVProcess.EntityJob.PGL) {
-          var job = ffxiv_.GetMonk();
-          if (job != null) {
-            if (send || !job.Equals(notify_state_.mnk)) {
-              notify_state_.mnk = job;
-              var e = new JSEvents.PlayerChangedEvent(player);
-              e.jobDetail = new JSEvents.PlayerChangedEvent.MonkDetail(job);
-              OnPlayerChanged(e);
-            }
-          }
-        } else if (player.job == FFXIVProcess.EntityJob.MCH) {
-          var job = ffxiv_.GetMachinist();
-          if (job != null) {
-            if (send || !job.Equals(notify_state_.mch)) {
-              notify_state_.mch = job;
-              var e = new JSEvents.PlayerChangedEvent(player);
-              e.jobDetail = new JSEvents.PlayerChangedEvent.MachinistDetail(job);
-              OnPlayerChanged(e);
-            }
-          }
-        } else if (player.job == FFXIVProcess.EntityJob.AST) {
-          var job = ffxiv_.GetAstrologian();
-          if (job != null) {
-            if (send || !job.Equals(notify_state_.ast)) {
-              notify_state_.ast = job;
-              var e = new JSEvents.PlayerChangedEvent(player);
-              e.jobDetail = new JSEvents.PlayerChangedEvent.AstrologianDetail(job);
-              OnPlayerChanged(e);
-            }
-          }
-        } else if (player.job == FFXIVProcess.EntityJob.SAM) {
-          var job = ffxiv_.GetSamurai();
-          if (job != null) {
-            if (send || !job.Equals(notify_state_.sam)) {
-              notify_state_.sam = job;
-              var e = new JSEvents.PlayerChangedEvent(player);
-              e.jobDetail = new JSEvents.PlayerChangedEvent.SamuraiDetail(job);
-              OnPlayerChanged(e);
+        var job = ffxiv_.GetJobSpecificData(player.job);
+        if (job != null) {
+          if (send || !job.Equals(notify_state_.jobData)) {
+            notify_state_.jobData = job;
+            var e = new JSEvents.PlayerChangedEvent(player);
```

Comment:
So one thing you're losing with this change is that there's no longer an equals check, so we'll send job data constantly.  Is it possible to just compare the bytes that we read and turned into job specific data and see if those changed before sending?

Reply:
There is a check on line 375 still unless I'm misunderstanding?

Reply:
Looking at it further I've replaced that line with `if (send || !JObject.DeepEquals(job, notify_state_.jobData)) {`.